lib/resty/core/semaphore.lua

```diff
@@ -0,0 +1,116 @@
+
```

Comment:
I suggest using the `ngx.semaphore` module name instead of `resty.core.semaphore`.